game-of-life-native/Game of Life/GameOfLifeViewController.swift

```diff
@@ -11,59 +11,104 @@ import Foundation
+
+        controlButton.frame = CGRect(
+                x: 0,
+                y: yOffset,
+                width: self.view.bounds.width / 2,
+                height: yOffset)
+
+        patternSelectionButton.frame = CGRect(
+                x: self.view.bounds.width / 2,
+                y: yOffset,
+                width: self.view.bounds.width / 2,
+                height: yOffset)
+        patternSelectionButton.setTitle("Choose a Pattern", for: .normal)
+
+        boardView.backgroundColor = .black
+        boardView.frame = CGRect(
+                x: 0,
+                y: controlButton.bounds.height + yOffset,
+                width: self.view.bounds.width,
+                height: self.view.bounds.width)
+    }
+
     func renderControlButtonLabel(controlButtonLabel: String) {
-        controllButton.setTitle(controlButtonLabel, for: .normal)
+        controlButton.setTitle(controlButtonLabel, for: .normal)
     }
-    
+
     func renderPatternSelectionVisibility(visibility: Bool) {
-        
+        patternSelectionButton.isHidden = !visibility
     }
-    
+
     func renderBoardWith(boardViewInput: KGOLBoardViewInput) {
         boardView.renderBoard(boardViewInput: boardViewInput)
     }
-    
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        if (appPresenter == nil){
-            let model = KGOLAppModel()
-            appPresenter = KGOLAppPresenter(model: model)
-            controllButton.backgroundColor = .green
-            controllButton.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
-            self.view.addSubview(controllButton)
-            
-            boardView.backgroundColor = .blue
-            self.view.addSubview(boardView)
-        }
-        
-        appPresenter?.bind(view: self)
-        boardView.willAppear()
+        view.addSubview(controlButton)
+        view.addSubview(patternSelectionButton)
+        view.addSubview(boardView)
     }
-    
-    @objc func buttonAction(sender: UIButton!) {
-        onControlButtonClicked()
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        setupLayout()
+        appPresenter.bind(view: self)
+        boardPresenter.bind(boardView: boardView)
     }
-    
+
+    @objc func controlButtonAction(sender: UIButton!) {
+        let _ = onControlButtonClicked()
+    }
+
+    @objc func patternSelectionButtonAction(sender: UIButton!) {
+        let alert = UIAlertController(title: "Patterns", message: "Please Choose a Pattern", preferredStyle: .actionSheet)
+
+        let patterns = KGOLPatternRepositoryCompanion().patterns() as NSArray
+        for pattern in patterns {
+            let patternEntity: KGOLPatternEntity = pattern as! KGOLPatternEntity
+
+            alert.addAction(UIAlertAction(title: patternEntity.getName(), style: .default, handler: { (action) in
+                let _ = self.onPatternSelected(patternEntity)
```

Comment:
To avoid weird crashes on closures is always good to specify that the controller aka `self` might be nil sometimes (if it has been discarded while waiting for the closure to finish . 

```
handler: { [weak self] action in 
   let _ = self?.onPatternSelected(patternEntity) 
```
That way if the controller is nil when we try to do pattern selected it won't crash the app, it will just don't do anything